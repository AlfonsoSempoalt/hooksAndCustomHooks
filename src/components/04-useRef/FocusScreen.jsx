import React from 'react';
import '../02-UseEffect/effects.css';

const FocusScreen = () => {
  const handleOnSubmit = (e) => {
    document.getElementById('name').select();
  };

  return (
    <>
      <h1>FocusScreen</h1>
      <hr />
      <input
        type='text'
        className='form-control mb-3'
        placeholder='Enter your name'
        id='name'
        />
      <input
        type='button'
        className='btn btn-outline-primary mb-2'
        value='focus'
        onClick={handleOnSubmit}
      />
    </>
  );
};

export default FocusScreen;
