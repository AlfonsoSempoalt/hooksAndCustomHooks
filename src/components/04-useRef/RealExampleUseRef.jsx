import React, { useState } from 'react';
import '../02-UseEffect/effects.css';
import MultipleCustomHooks from '../03-useFetch/MultipleCustomHooks';

export const RealExampleUseRef = () => {
    const [show, setShow] = useState(false);

    const handleOnClickShow = () => {
        setShow(!show);
    };

    return (
        <>
            <h1>useRef</h1>
            <hr />
            <button
                className='btn btn-primary mt-5'
                onClick={handleOnClickShow}
                >
                show/hide
            </button>
            {show && <MultipleCustomHooks />}
        </>
    );
};
