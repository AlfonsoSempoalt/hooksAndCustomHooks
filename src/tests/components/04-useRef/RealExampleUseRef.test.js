import React from 'react';
import '@testing-library/jest-dom';
import { expect, describe } from '@jest/globals';
import { shallow } from 'enzyme';
import { RealExampleUseRef } from '../../../components/04-useRef/RealExampleUseRef';

describe('tests in <RealExampleUseRef />', () => {
    let wrapper = shallow(<RealExampleUseRef />);

    beforeEach(() => {
        wrapper = shallow(<RealExampleUseRef />);
    });

    test('should match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });

    test('should show the component  <MultipleCustomHooks />', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
});
