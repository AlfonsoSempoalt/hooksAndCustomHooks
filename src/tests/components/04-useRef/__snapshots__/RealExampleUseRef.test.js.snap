// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <RealExampleUseRef /> should match with the snapshot 1`] = `
<Fragment>
  <h1>
    useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    show/hide
  </button>
</Fragment>
`;
